kolibri/core/analytics/utils.py

```diff
@@ -0,0 +1,115 @@
+        "l": max(usersess_agg["last"], contsess_agg["last"]).strftime("%Y-%m-%d"),  # last
+        "ss": summarylogs.count(),  # summ_started
+        "sc": summarylogs.exclude(completion_timestamp=None).count(),  # summ_complete
+        "sk": sesslogs_by_kind,  # sess_kinds
+        "lec": Lesson.objects.filter(dataset_id=dataset_id).count(),  # lesson_count
+        "ec": Exam.objects.filter(dataset_id=dataset_id).count(),  # exam_count
+        "elc": ExamLog.objects.filter(dataset_id=dataset_id).count(),  # exam_log_count
+        "alc": AttemptLog.objects.filter(dataset_id=dataset_id).count(),  # att_log_count
+        "ealc": ExamAttemptLog.objects.filter(dataset_id=dataset_id).count(),  # exam_att_log_count
+        "suc": contsessions_user.count(),  # sess_user_count
+        "sac": contsessions_anon.count(),  # sess_anon_count
+        "sut": int((contsessions_user.aggregate(total_time=Sum("time_spent"))["total_time"] or 0) / 60),  # sess_user_time
+        "sat": int((contsessions_anon.aggregate(total_time=Sum("time_spent"))["total_time"] or 0) / 60),  # sess_anon_time
+    }
+
+
+def extract_channel_statistics(channel):
+
+    channel_id = channel.id
+    tree_id = channel.root.tree_id
+
+    sessionlogs = ContentSessionLog.objects.filter(channel_id=channel_id, time_spent__lt=3600 * 2)
+    summarylogs = ContentSummaryLog.objects.filter(channel_id=channel_id)
+
+    sesslogs_by_kind = sessionlogs.order_by("kind").values("kind").annotate(count=Count("kind"))
+    sesslogs_by_kind = {log["kind"]: log["count"] for log in sesslogs_by_kind}
+
+    pop = list(sessionlogs.values("content_id").annotate(count=Count("id")).order_by("-count")[:50])
+
+    localfiles = LocalFile.objects.filter(available=True, files__contentnode__tree_id=tree_id).distinct()
+
+    contsessions_user = sessionlogs.exclude(user=None)
+    contsessions_anon = sessionlogs.filter(user=None)
+
+    return {
+        "ci": channel_id[:10],  # channel_id
+        "v": channel.version,  # version
+        "u": channel.last_updated.strftime("%Y-%m-%d") if channel.last_updated else None,  # updated
+        "pi": [item["content_id"][:10] for item in pop],  # popular_ids
+        "pc": [item["count"] for item in pop],  # popular_counts
+        "s": (localfiles.aggregate(Sum("file_size"))["file_size__sum"] or 0) / (2 ** 20),  # storage
+        "ss": summarylogs.count(),  # summ_started
+        "sc": summarylogs.exclude(completion_timestamp=None).count(),  # summ_complete
+        "sk": sesslogs_by_kind,  # sess_kinds
+        "suc": contsessions_user.count(),  # sess_user_count
+        "sac": contsessions_anon.count(),  # sess_anon_count
+        "sut": int((contsessions_user.aggregate(total_time=Sum("time_spent"))["total_time"] or 0) / 60),  # sess_user_time
+        "sat": int((contsessions_anon.aggregate(total_time=Sum("time_spent"))["total_time"] or 0) / 60),  # sess_anon_time
+    }
```

Comment:
Maybe move the comments above the items being returned to make it easier to scan?